api/v1/logic/cms/utils.js

```diff
@@ -0,0 +1,35 @@
+const { ERRORS } = require("../../constants");
+const { CMS_ERRORS } = require("./errors");
+const { LogicError } = require("../../utils/errors");
+
+const { generateToken } = require("./auth");
+
+async function fetchWithToken({method, url, data={}, retry=true}) {
+  if (!global.cmsToken) {
+    global.cmsToken = await generateToken();
+  }
+  const header = {
+    "Authorization": global.cmsToken,
+  };
+  const body = await method(url, data, header);
```

Comment:
if we do not reuse `header` then no need to create a new variable for it
```suggestion
  const body = await method(url, data,  {
    "Authorization": global.cmsToken,
  });
```